admin-panel/src/main/java/net/cryptic_game/backend/admin/service/server_management/EndpointServiceImpl.java

```diff
@@ -0,0 +1,135 @@
+    }
+
+    @Override
+    public Optional<DisabledEndpoint> findDisabledEndpoint(final String path) {
+        return this.disabledEndpointRepository.findById(path)
+                .map(this.endpointConverter::toDto);
+    }
+
+    @Override
+    public Mono<Endpoint> disableEndpoint(final String path, final DisabledEndpoint disabledEndpoint) {
+        if (this.disabledEndpointRepository.findById(path).isPresent()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "already disabled");
+        }
+        return this.serverCommunication.responseFromServer("/admin_panel/disable",
+                new JSONObject(Map.of("id", path)))
+                .map(response -> {
+                    try {
+                        final DisabledEndpointModel model = this.endpointConverter.toModel(disabledEndpoint);
+                        model.setEndpointPath(path);
+                        this.disabledEndpointRepository.save(model);
+                        final Endpoint endpoint = this.objectMapper.readValue(response, Endpoint.class);
+                        endpoint.setReason(disabledEndpoint.getReason());
+                        return endpoint;
+                    } catch (JsonProcessingException e) {
+                        e.printStackTrace();
+                        throw new InternalServerErrorException("Invalid Format from server to admin-panel");
+                    }
+                })
+                .doOnError(err -> {
+                    if (err instanceof NotFoundException) {
+                        throw new NotFoundException(path, "endpoint not found");
+                    }
+                });
+    }
+
+    @Override
+    public Mono<Endpoint> enableEndpoint(final String path) {
+        if (this.disabledEndpointRepository.findById(path).isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "not disabled");
+        }
+        return this.serverCommunication.responseFromServer("/admin_panel/enable",
+                new JSONObject(Map.of("id", path)))
+                .map(response -> {
+                    try {
+                        this.disabledEndpointRepository.deleteById(path);
+                        return this.objectMapper.readValue(response, Endpoint.class);
+                    } catch (JsonProcessingException e) {
+                        e.printStackTrace();
+                        throw new InternalServerErrorException("Invalid Format from server to admin-panel");
+                    }
+                })
+                .doOnError(err -> {
+                    if (err instanceof NotFoundException) {
+                        throw new NotFoundException(path, "endpoint not found");
+                    }
+                });
+    }
+
+    @Override
+    public Mono<Endpoint> endpointInfo(final String path) {
+        return this.serverCommunication.responseFromServer("/admin_panel/info",
+                new JSONObject(Map.of("id", path)))
```

Comment:
Use the object mapper to convert a object to a json string. But if you use the spring we client, you could pass in a object directly.